perfkitbenchmarker/linux_benchmarks/specsfs2014_benchmark.py

```diff
@@ -0,0 +1,264 @@
+    see http://www.spec.org/sfs2014/.
+  vm_groups:
+    clients:
+      vm_spec: *default_single_core
+      vm_count: null
+    gluster_servers:
+      vm_spec: *default_single_core
+      disk_spec: *default_500_gb
+      vm_count: 3
+"""
+
+SPEC_SFS_2014_ISO = 'SPECsfs2014_SP1.iso'
+SPEC_SFS_2014_LICENSE = 'netmist_license_key'
+SPEC_DIR = 'spec'
+
+VOLUME_NAME = 'gv0'
+MOUNT_POINT = '/scratch'
+PUBLISHED_METRICS = frozenset([
+    'achieved rate', 'average latency', 'overall throughput',
+    'read throughput', 'write throughput'
+])
+METADATA_KEYS = frozenset([
+    'op rate', 'run time', 'processes per client', 'file size',
+    'client data set size', 'starting data set size', 'initial file space',
+    'maximum file space'
+])
+
+
+def GetConfig(user_config):
+  return configs.LoadConfig(BENCHMARK_CONFIG, user_config, BENCHMARK_NAME)
+
+
+def CheckPrerequisites():
+  """Verifies that the required resources are present.
+
+  Raises:
+    perfkitbenchmarker.data.ResourceNotFound: On missing resource.
+  """
+  data.ResourcePath(SPEC_SFS_2014_ISO)
+  data.ResourcePath(SPEC_SFS_2014_LICENSE)
+  if FLAGS.specsfs2014_config:
+    data.ResourcePath(FLAGS.specsfs2014_config)
+
+
+def _PrepareClient(vm, gluster_server):
+  """Prepares a SPEC client by copying SPEC to the VM and mounting Gluster."""
+  mount_dir = 'spec_mnt/'
+  vm.RemoteCommand('mkdir -p %s' % mount_dir)
+  vm.RemoteCommand('mkdir -p %s' % SPEC_DIR)
+  vm.PushFile(data.ResourcePath(SPEC_SFS_2014_ISO))
+  vm.PushFile(data.ResourcePath(SPEC_SFS_2014_LICENSE), SPEC_DIR)
+  vm.RemoteCommand('sudo mount -t iso9660 -o loop %s %s' %
+                   (SPEC_SFS_2014_ISO, mount_dir))
+  vm.RemoteCommand('cp -r %s/* %s' % (mount_dir, SPEC_DIR))
+  vm.RemoteCommand('sudo umount {0} && sudo rm -rf {0}'.format(mount_dir))
+  if gluster_server:
+    vm.Install('gluster')
+    gluster.MountGluster(vm, gluster_server, VOLUME_NAME, MOUNT_POINT)
+
+
+def _PrepareServer(vm):
+  """Prepares a Gluster server by installing GlusterFS."""
+  vm.Install('gluster')
+
+
+def _ConfigureSpec(prime_client, clients):
+  """Configures SPEC SFS 2014 on the prime client."""
+  config_path = posixpath.join(SPEC_DIR, 'sfs_rc')
+
+  if FLAGS.specsfs2014_config:
+    prime_client.PushFile(
+        data.ResourcePath(FLAGS.specsfs2014_config), config_path)
+  else:
+    stdout, _ = prime_client.RemoteCommand('pwd')
+    exec_path = posixpath.join(stdout.strip(), SPEC_DIR, 'binaries',
+                               'linux', 'x64', 'netmist')
+    configuration_overrides = {
+        'USER': prime_client.user_name,
+        'EXEC_PATH': exec_path.replace('/', '\/'),
+        'CLIENT_MOUNTPOINTS': 'mountpoints.txt',
+        'BENCHMARK': FLAGS.specsfs2014_benchmark,
+        'LOAD': FLAGS.specsfs2014_load,
+        'NUM_RUNS': FLAGS.specsfs2014_num_runs,
+        'INCR_LOAD': FLAGS.specsfs2014_incr_load
+    }
+
+    sed_expressions = ' '.join(
+        ['-e "s/{0}=.*/{0}={1}/"'.format(k, v)
```

Comment:
Yes, but with the exception of the path, none of the other variables will contain characters that will confuse sed - I've added a comment explaining this